pm/sendermonitor_test.go

```diff
@@ -185,32 +185,43 @@ func TestQueueTicketAndSignalNewBlock(t *testing.T) {
 			ClaimedInCurrentRound: big.NewInt(0),
 		},
 	}
+
+	ts := newStubTicketStore()
 	smgr.claimedReserve[addr] = big.NewInt(100)
-	sm := NewSenderMonitor(claimant, b, smgr, tm, 5*time.Minute, 3600)
+	sm := NewSenderMonitor(claimant, b, smgr, tm, ts, 5*time.Minute, 3600)
 	sm.Start()
 	defer sm.Stop()
 
 	assert := assert.New(t)
 
 	// Test queue ticket
+	// test fail
+	ts.storeShouldFail = true
+	assert.EqualError(sm.QueueTicket(addr, defaultSignedTicket(addr, uint32(0))), "stub TicketStore store error")
+	ts.storeShouldFail = false
 
-	sm.QueueTicket(addr, defaultSignedTicket(uint32(0)))
+	err := sm.QueueTicket(addr, defaultSignedTicket(addr, uint32(0)))
+	assert.Nil(err)
 	time.Sleep(20 * time.Millisecond)
-
-	assert.Equal(sm.(*senderMonitor).senders[addr].queue.Length(), int32(1))
+	qlen, err := sm.(*senderMonitor).senders[addr].queue.Length()
+	assert.Nil(err)
+	assert.Equal(qlen, 1)
 
 	qc := &queueConsumer{}
-	go qc.Wait(1, sm)
+	done := make(chan struct{})
+	go qc.Wait(1, sm, done)
 
 	tm.blockNumSink <- big.NewInt(5)
-	time.Sleep(20 * time.Millisecond)
-
+	<-done
 	// check that ticket is now removed from queue
-	assert.Equal(sm.(*senderMonitor).senders[addr].queue.Length(), int32(0))
+	time.Sleep(time.Millisecond)
```

Comment:
Let's make this longer (i.e. at least 20) since we know CI can be flaky especially with short sleeps.